### PR TITLE
Fixing urlscan bug

### PR DESCRIPTION
**Pull request by hod-alpert**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes https://github.com/demisto/etc/issues/31551

## Description
This PR fixes the urlscan bug:
- Removes two endless loops
- Uses api-key in all requests to enjoy the extended rate limit
- Added logs for better debugging

## Minimum version of Demisto
- [x] 5.0.0
- [ ] 5.5.0
- [ ] 6.0.0

## Does it break backward compatibility?
   - [x] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

**Comment by hod-alpert** [comment]
@Itay4 Ready for review

File: Packs/UrlScan/ReleaseNotes/1_0_6.md [hunk 1]
```diff
@@ -0,0 +1,4 @@
+
+#### Integrations
+##### urlscan.io
+- Fixed a bug where exceeding quota limit caused timeout exception
```
Comment (amshamah419) [inline Packs/UrlScan/ReleaseNotes/1_0_6.md:4]:
```suggestion
- Fixed a bug where exceeding the quota limit caused a timeout exception.
```

File: Packs/UrlScan/Integrations/UrlScan/UrlScan.py [hunk 2]
```diff
@@ -41,6 +41,8 @@ def http_request(method, url_suffix, json=None, wait=0, retries=0):
             'Content-Type': 'application/json',
             'Accept': 'application/json'
         }
+    LOG('requesting https request with method: {}, url: {}, data: {}'.format(method, BASE_URL + url_suffix, json))
```
Comment (Itay4) [inline Packs/UrlScan/Integrations/UrlScan/UrlScan.py:44]: use `demisto.debug` instead

File: Packs/UrlScan/Integrations/UrlScan/UrlScan.py [hunk 3]
```diff
@@ -66,7 +68,7 @@ def http_request(method, url_suffix, json=None, wait=0, retries=0):
 
         response_json = r.json()
         error_description = response_json.get('description')
-        should_continue_on_blacklisted_urls = demisto.args().get('continue_on_blacklisted_urls')
+        should_continue_on_blacklisted_urls = argToBoolean(demisto.args().get('continue_on_blacklisted_urls', False))
```
Comment (Itay4) [inline Packs/UrlScan/Integrations/UrlScan/UrlScan.py:71]: add `true,false` as predefined values to the arg with `defaultvalue` as `false`

File: Packs/UrlScan/Integrations/UrlScan/UrlScan.py [hunk 4]
```diff
@@ -120,32 +118,39 @@ def is_truthy(val):
...
-    max_time = time.time() + timeout if timeout else None
+    max_time = time.time() + timeout
     tries = 0
-
-    last_item = None
+    # According to the doc - The most efficient approach would be to wait at least 10 seconds before starting to poll
+    time.sleep(10)
     while True:
-
+        LOG('Number of Polling attempts: {}'.format(tries))
```
Comment (Itay4) [inline Packs/UrlScan/Integrations/UrlScan/UrlScan.py:133]: use `demisto.debug` instead

File: Packs/UrlScan/Integrations/UrlScan/UrlScan.py [hunk 5]
```diff
@@ -549,28 +550,28 @@ def format_http_transaction_list():
...
-        demisto.results(get_result_page())
-    if demisto.command() == 'urlscan-poll-uri':
-        poll_uri()
-
-
-except Exception as e:
-    LOG(e)
-    LOG.print_log(False)
-    return_error(e.message)
+if __name__ in ('__main__', '__builtin__', 'builtins'):
```
Comment (Itay4) [inline Packs/UrlScan/Integrations/UrlScan/UrlScan.py:553]: use `main()`

File: Packs/UrlScan/Integrations/UrlScan/Urlscan_test.py [hunk 6]
```diff
@@ -29,3 +31,13 @@ def test_continue_on_blacklisted_error_arg(mocker, requests_mock, continue_on_bl
...
+
+
+def test_endless_loop_on_failed_response(requests_mock, mocker):
+    from UrlScan import format_results
+    mocker.patch(RETURN_ERROR_TARGET)
+
+    with open('./test_data/capitalne.json', 'r') as f:
+        response_data = json.loads(f.read())
+    requests_mock.get(RESULT_URL + 'uuid', status_code=200, json=response_data)
+    format_results('uuid')
```
Comment (Itay4) [inline Packs/UrlScan/Integrations/UrlScan/Urlscan_test.py:43]: assert results

File: Packs/UrlScan/Integrations/UrlScan/Urlscan_test.py [hunk 7]
```diff
@@ -29,3 +31,13 @@ def test_continue_on_blacklisted_error_arg(mocker, requests_mock, continue_on_bl
         assert return_error_mock.call_count == 0
     else:
         assert return_error_mock.call_count == 1
+
+
+def test_endless_loop_on_failed_response(requests_mock, mocker):
```
Comment (Itay4) [inline Packs/UrlScan/Integrations/UrlScan/Urlscan_test.py:36]: add test docstring

File: Packs/UrlScan/Integrations/UrlScan/Urlscan_test.py (hunk 6 above)
Comment (hod-alpert) [reply Packs/UrlScan/Integrations/UrlScan/Urlscan_test.py:43]: All this method does is demisto_results, it does not return anything, not sure what to assert

**Comment by Itay4** [comment]
@hod-alpert please unskip the test

**Review by Itay4 — APPROVED** [review]

**Pull request merged**